### PR TITLE
[Woo POS] Add totals section to the "Payment Successful" screen

### LINKED ISSUE #11759: [Woo POS] Add totals section to the "Payment successful" screen
We need to enhance the "Payment Successful" screen by adding a detailed totals section. This section will display a summary of the order totals, including the subtotal, tax, and final total. 

Branch off of this PR by @samiuelson which actually adds the payment success screen:
https://github.com/woocommerce/woocommerce-android/pull/11749

Should be completed before [this](https://github.com/orgs/woocommerce/projects/247/views/3?pane=issue&itemId=68674795) task for payment failure (which is not part of M1 "happy path")

Steps to Reproduce

1. After successful payment the app should redirect to the "Payment successful" screen.
1. Observe the "Payment Successful" screen.

Expected Outcome

The "Payment Successful" screen should display a detailed section with the transaction totals, including subtotal, tax, and final total.

UI from Figma: 

Image

```[tasklist]
### Tasks
- [x] Fetch the necessary order details from the database once payment is completed.
- [x] Ensure the order data is mapped to the UI state correctly.
- [x] Update the composable function for the "Payment Successful" screen to include and display the totals section.
- [x] Display the subtotal, tax, and final total prominently on the screen. This allows users to verify the amounts charged and ensures clarity around the transaction.
- [ ] https://github.com/woocommerce/woocommerce-android/issues/11828
```

### PR DESCRIPTION
**Pull request by backwardstruck**
#### Closes: 
#11759

### Description
This PR addresses GitHub issue #11759 by adding a detailed totals section to the "Payment Successful" screen. The enhancement provides users with a transparent summary of their transaction, including subtotal, tax, and final total. 

Branched from https://github.com/woocommerce/woocommerce-android/pull/11749

### Steps to Reproduce
1. Initiate and complete a transaction through POS.
2. After a successful payment, the app should redirect to the "Payment Successful" screen.
3. Observe the displayed transaction details on the "Payment Successful" screen.

### Testing Information
- **Devices Used:** Tested on Android tablet emulator.
- **Affected Areas:** The change affects the "Payment Successful" screen and related view models.
- **Critical Flows:** Ensure that the new transaction button works as expected, returning the user to the new transaction screen.
- **Areas Not Tested:** This PR does not modify anything outside of the UI changes on the “Payment Successful” screen.
- **Remaining Unknowns:** None identified at this moment.

### Images/GIF
Include before and after images/GIFs when appropriate.

- **Before:** 

Screenshot_20240626_184349

- **After:** 

Screenshot_20240627_153718

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

---

Feel free to adjust any sections as necessary to match the specifics of your work.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt [hunk 1]
```diff
@@ -77,6 +79,22 @@ fun WooPosPaymentSuccessScreen(
                     fontWeight = FontWeight.SemiBold,
                     color = MaterialTheme.colors.onSurface,
                 )
+                Spacer(modifier = Modifier.height(16.dp))
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt]: I briefly checked the code, and everything seems correct except for one thing. There should be a border around this, and the colors and sizes of the fonts are different. Please take a look at Figma.

image

Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt]: Thanks! It should look more like the design now.

**Comment by backwardstruck** [comment]
FYI @kidinov I made a follow up ticket to match the design exactly:
https://github.com/woocommerce/woocommerce-android/issues/11828

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/223">19.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by kidinov — APPROVED** [review]
@backwardstruck 👋

I've made a few small changes to the PR. Could you please take a look for knowledge sharing purposes? Feel free to change those in the next PR

image

**Comment by backwardstruck** [comment]
Thanks for those additional changes @kidinov. I've made notes on those so I don't make the same mistake next time!

I also fixed `Changed incorect text` in this [PR](https://github.com/woocommerce/woocommerce-android/pull/11827/commits). I should have mentioned that.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ class WooPosTotalsViewModel @Inject constructor(
     private val childrenToParentEventSender: WooPosChildrenToParentEventSender,
     private val cardReaderFacade: WooPosCardReaderFacade,
     private val orderDetailRepository: OrderDetailRepository,
-    private val currencyFormatter: CurrencyFormatter,
+    private val priceFormat: WooPosFormatPrice,
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:30]: I'll use this going forward.

**Pull request merged**